test/ModuleInterface/parameterized-protocol-type-inheritance.swift

```diff
@@ -0,0 +1,23 @@
+// RUN: %target-swift-emit-module-interface(%t/Fancy.swiftinterface) %s -module-name Library
+// RUN: %target-swift-typecheck-module-from-interface(%t/Fancy.swiftinterface) -module-name Library
+// RUN: %FileCheck %s < %t/Fancy.swiftinterface
+
+// CHECK: import Foundation
+import Foundation
+
+public protocol Fancy<Stuff> {
+    associatedtype Stuff
+}
+
+// CHECK: extension Swift.String : Library.Fancy {
+// CHECK: public typealias Stuff = CoreFoundation.CFString
+extension String: Fancy<CFString> {
+}
+
+public protocol Q {
+}
+
+// CHECK: public struct S : Library.Fancy & Library.Q
+public struct S: Fancy<Int> & Q {
```

Comment:
It would be nice to add a few more tests:
- Try something like `((Fancy<Int> & Q) & P)`
- More than one ParameterizedProtocolType in the inheritance clause
- Composition with a superclass. For example, we allow this today:

```
class Base {}

protocol P<A> {
  associatedtype A
}

class Derived: Base & P<Int> {}
```

- Composition with an inverse requirement. For example, we allow this:

```
protocol P<A>: ~Copyable {
  associatedtype A
}

struct S: P<Int> & ~Copyable {}
```